### PR TITLE
cnf-tests: Update OCP version to 4.16

### PR DESCRIPTION
**Pull request by liornoy**
After OCP v4.15 branched out, we update the OCP version reference
to 4.16 in every place in the repository.

Note: This also switches the base images of cnf-tests and ztp
to RHEL9 with golang 1.21 support.

### CODE REVIEW

**Comment by liornoy** [comment]
done.
This changes ztp and the cnf-tests images.
cc @yanirq @lack

**Comment by yanirq** [comment]
I think this might be a good time to switch to RHEL 9 base images supporting golang 1.21:
`registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 `
NTO already [switched to golang 1.21 ](https://github.com/openshift/cluster-node-tuning-operator/pull/879)

**Comment by liornoy** [comment]
> `registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 `

Done.

File: cnf-tests/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.15 AS builder-stresser
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16  AS builder-stresser
```
Comment (SchSeba) [inline cnf-tests/Dockerfile:1]: can you please remove the double space?

Comment (liornoy) [reply cnf-tests/Dockerfile]: done. thanks!

**Review by SchSeba — APPROVED** [review]
/lgtm
/approve

**Pull request merged**